### ISSUE TITLE
Suggestion: different scoring system

### ISSUE THREAD

**Issue by Gliptal** [eezstreet/SWATEliteForce#209]
Currently, the scoring system is based on the assumption that 100/100 is the _ideal_ mission outcume, mainly involving no suspects injured or killed. The issue with this paradigm is it assumes that everything is in control of the player, and that a 100/100 is always achievable no matter what the perps AI's behaviour is. For example, suppose the squad makes an entry into a room, and before the flashbang goes off the player pops a suspect pointing a gun to an hostage. With the current score system that (lawful) sequence of events means I can no longer get 100/100: the player _is_ penalized even though he reacted organically and correctly to the situation at hand. There are plenty of similar situations where the player is forced into doing actions that make a perfect score impossible.

These small score deficiencies quickly add-up during mission (especially longer ones), and especially when playing on Elite it feels like advancing to the next mission is more of an exercise in repeating the current one until the AI, positioning, and contingencies all align and permit an ideal execution. I found myself restarting a mission simply because I know I couldn't get a 90/100 after downing the nth suspect, making for a frustrating repeat when I know I reacted correctly to the situation at hand.

My proposal is to assume that 100/100 is the score given not to the perfect mission, but to the perfect execution under the circumstances. As such, instead of counting up from 40 and adding a set value for every item, count down from 100 and subtract for every incorrect behaviour. e.g. missing TOC calls and not securing evidence would detract from the score, whereas lawfully downing a suspect would not.

**Comment by kevinfoley** [comment]
Your suggestion doesn't require counting down from 100, it just requires scoring all suspects the same regardless of whether they were arrested, incapacitated, or killed.

This is something I often find myself debating. For the first 2/3rds of the campaign, it is easy to get a perfect score by equipping the entire team with less-lethal weapons. In later missions where there is a lot of glass and/or long distance shooting, it becomes impractical to use less-lethal weapons. The AI is also overzealous and will happily kill suspects who are stunned or running away.

On the other hand, the ideal scenario for any police force is always to arrest all suspects with no casualties. The department might be satisfied if all threatening suspects end up in the hospital or the morgue, but they'll probably be _happier_ if the suspects all end up in prison and face justice through the courts. In reality, every time a cop seriously wounds or kills a suspect, there has to be an investigation, and usually the officer is placed on mandatory paid leave while the shooting is reviewed (not sure if this applies to SWAT). Of course, there isn't a great way to reflect this in the game except with the scoring system. What would we do, force the player to fill out paperwork and make them wait for 15 minutes before they can start the next level?

As I've gotten better at the game, I've found it easier to get those scores in the 90 - 100 range even on Elite. A major factor is in your deployment of distraction devices - a well-placed flashbang or CS grenade can disable a roomful of suspects long enough for you to punch the guns out of their hands and arrest them. On the other hand, I've sometimes found myself using questionable strategies like immediately shooting any non-threatening suspect in the arm to get him to drop his weapon. 

Overall I have mixed feelings about the scoring. It encourages you to do your best to arrest people without killing them, but it also can cause you to play SEF like a game instead of making decisions based on officer and public safety. I'm not sure there is any better solution.

**Comment by Gliptal** [comment]
Your last point is exactly what I've been feeling: sometimes to get more than 90 points I end up making gamey decisions rather than intelligent organic ones.

One easy way around this is to add a difficulty level, or make it so that you get all the additional perps and traps and so on on Hard as well. This lowers the threshold enough that you can get past the missions without the ideal run, but keeps them hard and engaging.

**Comment by jose21crisis** [comment]
I agree with the fact that the score is "crooked", you get detracted points for lawful things. But, the score system from the base is the PUBLIC's look on your performance, not the DEPARTMENT's. Which means that the public expects what to bring everyone in custody, with all evidence secure and everyone healthy, including team members. 

There is something I suggested some time ago, and it was changing the score system to work like SWAT 2's. This works this way: On each mission the player gets a pre-set starting score (75) and each action the officers do affects that score: Arresting or incapacitating a threatening suspect (+2), rescuing a civlian (+4), securing evidence (+5) gave points. Killing suspects gave no points or penalties. Friendly fire (-5 for hit, -15 for incap), losing officers to suspect fire (-5), injuring/killing non-threatening suspects (-5 IIRC), allowing a suspect to injure/kill a civilian (-10/-20) and injuring or killing civilians (-20/-40) were heavy penalties. And the score was used for calculating the player's performance. Say, on mission 1 player obtained 100 and obtained 150 in mission 2, the player's overall performance was 125, but if then the player screws up on mission 3 and obtains 50, the overall performance is lowered to 100. If the score gets below a certain point, the Chief of Police (AKA the player) was fired, terminating the campaign.

Not sure if it can be done. But'll look into it.

**Comment by eezstreet** [comment]
Personally, I think the scoring system is mostly fine. If you manage to get a 100 on a mission, you should (rightly) be considered a super elite operator that is able to rescue an entire building of people. You're almost heroic in a sense.

The problem I think with the scoring system (and indeed, what makes you "think of it as a game") is that it's entirely difficulty based on whether you pass or fail. In the original game (not SEF), you had easy mode, which was impossible to fail (unless you died or failed a mission objective), normal mode, which is difficult to fail, hard, which is really about average, and elite, which is *brutal* to complete. 

In the original game, there's no difference between scoring a 95 on Easy and scoring a 95 on Elite. Absolutely none. Both pass you the mission, and both are (more or less) equally difficult to accomplish. That's why I think difficulty is deceptive, and saying "oh, I got a 95 on Elite" should be a lot more of an accomplishment than "oh, I got a 95 on Easy". But it's not really, currently.

I think the player should consider the game like a test, because that's how I originally envisioned the mod. Here's what I would recommend changing:

 - Either remove the difference in scoring requirements or make them very small (60 on Easy, 65 on Normal, 70 on Hard and 75 on Elite)
 - Make spawns *much* more based on difficulty level. Hard is pretty much fine where it is, but Elite mode should spawn more stuff and the lower difficulties should spawn less.
 - Make the penalty amount based on the difficulty level. Unauthorized Use of Deadly Force should penalize -20 on Elite, -15 on Hard, and -10 on lower difficulties.
 - Add some random, scripted stuff that varies based on difficulties. For instance, Louie Baccus might be packing heat on higher difficulties, at random. Add some scripted enemies that show up on A-Bomb and start shooting at rivals, etc. More dynamic and unexpected stuff that makes the player think on their feet more often.
 - Make the score less about how a suspect was neutralized, and make it more about how many civilians and officers were unharmed.

There are my thoughts, but it can go any way.

**Comment by kevinfoley** [comment]
My concern with increasing the number of spawns on Elite is that the game is already extremely unrealistic in terms of the number of suspects per mission. When do you hear about SWAT engaging teams of 15 to 30 suspects in paramilitary gear? Never, at least not in the US. The number of suspects in the game right now is a little silly, though I understand it needs to be this way to keep the game interesting - nobody wants to spend half an hour clearing a 4-story diamond center that only has 4 perps. 

A few more ideas for the "random, scripted stuff" that I think would work well to increase the difficulty on Elite, based on real-life scenarios that police encounter:

- Active shooter (i.e. shooting spree) - when the level starts, suspect(s) are actively hunting and gunning down civilians. This would work best with a much higher civilian count.
- Bullet hose - right now suspects are very frugal with their ammunition. That certainly isn't the case in real life - career criminals tend to spray magazine after magazine in the direction of the police, with no concern for collateral damage. One of the reasons the North Hollywood shootout lasted for so long was because the suspects had assault rifles with drum magazines and were putting down so much cover fire, the police had trouble approaching them. If suspects in SEF used more suppressive fire, we'd have to play more carefully and think more tactically
- Real hostages - in an actual hostage scenario, police have to seriously consider how to protect the lives of hostages. This can mean having to make a surprise entry and bring down an entire room of suspects in a couple of seconds before they have time to kill their captives. The suspects never kill the hostages in SWAT 4 - I've seen them point their guns at civilians a few times, but they never _fire_, so I'm never worried that some hostages are going to get executed if I make my entry too slowly.

**Comment by jose21crisis** [comment]
Ok, time to get de-cracking

- Yes, we need an active shooter scenario. No bad things about this one.
- Have you seen low skill suspects and wild gunners? They tend to go full auto for a fairly long time. Also, some suspects are military trained (or at least, highly trained). Those suspects should not go full auto wasting mag after mag dealing suppressive fire. Those guys know that every round is useful.
- Ok.......we need to make suspects take less time to shoot the hostages and make them more likely to do that. Bam! Problem solved. Also, Insane suspects.

**Comment by kevinfoley** [comment]
I have not noticed many suspects going full auto for a long amount of time.

You are wrong about how training should correlate to suppressive fire. [Suppressive fire ](https://en.wikipedia.org/wiki/Suppressive_fire)is a very common and _very_ legitimate tactic used by military and paramilitary forces. It helps with area denial, disrupting enemy operations and maneuvers, and generally making your opponents more stressed and less effective. Police can't use suppressive fire because of its general destructiveness and the unacceptable risks of collateral damage, but a hardened criminal isn't going to care if they cause thousands of dollars in property damage or kill innocent civilians while they're trying to keep the police pinned down.

Again, read about the North Hollywood shootout - the suspects had drum magazines and weren't afraid to use them, so it was difficult for the police to approach them.

**Comment by jose21crisis** [comment]
Yes, I am aware of the effectiveness of suppressive fire (Tom Clancy's Ghost Recon showed how effective it can be, specially when you are at the receiving end).

What I wanted to say (with the edit) is that high skill suspect could use suppressive fire, but not all the time. They would run out of ammo (they have 2 mags per weapon as of now), and they aren't too smart about reloading. 

The best thing we could do is actually make a Squad AI that makes some suspects delay the entry team while others go and flank them/execute hostages/plan ambush/flee the mission area, since right now there is only Individual AI. Also, make some suspects hold their ground instead of fleeing the team.

**Comment by kevinfoley** [comment]
If they only have two mags, can't we just give them more mags?

**Comment by jose21crisis** [comment]
Yes, we can. They still don't know what to do with those extra mags. Until they run dry and remember that they have to put a new one.

**Comment by kevinfoley** [comment]
Well, presumably it would be much easier to fix that than it would be to add squad AI for the suspects.

**Comment by Gliptal** [comment]
This got derailed a bit.
